### PR TITLE
[RFC] Add parser for phylogenetic tree and add OpenDialog for opening files in one place.

### PR DESCRIPTION
**Pull request by pvanagtmaal**

### CODE REVIEW

File: src/main/java/nl/tudelft/dnainator/ui/controllers/WindowController.java [hunk 1]
```diff
@@ -23,53 +26,58 @@
...
 	@FXML private View view;
-	private FileLoadService loadService;
+	private GraphLoadService graphLoadService;
+	private NewickLoadService newickLoadService;
 	private ProgressDialog progressDialog;
 
 	/**
 	 * Constructs a WindowController object, creating
-	 * a {@link FileLoadService} to go with it.
+	 * a {@link GraphLoadService} to go with it.
```
Comment (ohuang12) [inline src/main/java/nl/tudelft/dnainator/ui/controllers/WindowController.java]: And a NewickLoadService.

File: src/main/java/nl/tudelft/dnainator/ui/controllers/WindowController.java [hunk 2]
```diff
@@ -23,53 +26,58 @@
...
 	 */
 	@FXML
 	private void initialize() {
-		loadService = new FileLoadService();
+		graphLoadService = new GraphLoadService();
+		newickLoadService = new NewickLoadService();
```
Comment (ohuang12) [inline src/main/java/nl/tudelft/dnainator/ui/controllers/WindowController.java]: Shouldn't an ExceptionDialog be thrown when it fails loading in the newick file?

File: src/main/java/nl/tudelft/dnainator/ui/controllers/WindowController.java [hunk 3]
```diff
@@ -23,53 +26,58 @@
...
-		progressDialog = new ProgressDialog(root, loadService);
-		FileChooser chooser = new FileChooser();
-		chooser.setTitle("Open node file");
-		chooser.getExtensionFilters().add(
-				new ExtensionFilter("Graph files", "*.node.graph"));
-		File nodeFile = chooser.showOpenDialog(root.getScene().getWindow());
-		if (nodeFile == null) {
-			return;
+		progressDialog = new ProgressDialog(root, graphLoadService);
+		Optional<ButtonType> result = new OpenDialog(root, Alert.AlertType.NONE, graphLoadService,
```
Comment (ohuang12) [inline src/main/java/nl/tudelft/dnainator/ui/controllers/WindowController.java]: Is it really necessary to pass the AlertType to OpenDialog?

Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/controllers/WindowController.java]: It isn't, will fix.

File: src/main/java/nl/tudelft/dnainator/ui/controllers/WindowController.java (hunk 2 above)
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/controllers/WindowController.java]: Isn't that being done already?

File: src/main/java/nl/tudelft/dnainator/ui/controllers/WindowController.java (hunk 1 above)
Comment (pvanagtmaal) [inline src/main/java/nl/tudelft/dnainator/ui/controllers/WindowController.java]: Correct. Will update.

File: src/main/java/nl/tudelft/dnainator/ui/DNAinator.java [hunk 4]
```diff
@@ -29,6 +29,8 @@
 	public void start(Stage primaryStage) throws Exception {
 		primaryStage.setTitle(DNAINATOR);
 		primaryStage.getIcons().add(new Image(getClass().getResourceAsStream(ICON)));
+		primaryStage.setMinWidth(MIN_WIDTH);
```
Comment (gfokkema) [inline src/main/java/nl/tudelft/dnainator/ui/DNAinator.java]: Does this have to be in this commit?

**Comment by gfokkema** [comment]
Reviewed, will merge after update :).

**Comment by pvanagtmaal** [comment]
updated :)

File: src/main/java/nl/tudelft/dnainator/ui/services/NewickLoadService.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.concurrent.Service;
+import javafx.concurrent.Task;
+import nl.tudelft.dnainator.parser.TreeParser;
+import nl.tudelft.dnainator.tree.TreeNode;
+
+import java.io.File;
+
+/**
+ *
```
Comment (Unia) [inline src/main/java/nl/tudelft/dnainator/ui/services/NewickLoadService.java]: Missing JavaDoc.

File: src/main/java/nl/tudelft/dnainator/ui/widgets/dialogs/OpenDialog.java [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+		nodeField.setPrefWidth(TEXTFIELD_WIDTH);
+		nodeField.setPromptText("Node file (*" + NODE + ")");
+		nodeField.setOnMouseClicked(e -> browseNodeListener());
+
+		edgeField = new TextField();
+		edgeField.setPromptText("Edge file (*" + EDGE + ")");
+		edgeField.setOnMouseClicked(e -> browseEdgeListener());
+
+		newickField = new TextField();
+		newickField.setPromptText("Tree file (*" + NEWICK + ")");
```
Comment (Unia) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/dialogs/OpenDialog.java]: s/Tree/Newick/ ?

File: src/main/java/nl/tudelft/dnainator/ui/widgets/dialogs/OpenDialog.java [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+		grid.add(nodeField, 1, 0);
+		grid.add(edgeField, 1, 1);
+		grid.add(newickField, 1, 2);
+	}
+
+	private void setupLabels(GridPane grid) {
+		grid.add(new Label("Nodes:"), 0, 0);
+		grid.add(new Label("Edges:"), 0, 1);
+		grid.add(new Label("Tree:"), 0, 2);
```
Comment (Unia) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/dialogs/OpenDialog.java]: s/Tree/Newick ?

**Pull request merged**